doc/Coding Guidelines.md

```diff
@@ -0,0 +1,107 @@
+
+## ES2015
+We do preffer to use ES2015 constructions in the code - and there is a babel compiler set to allow us to do so. So use the ES modules instead of `require` and feel free to use other fancy structures.
+
+## Components
+Components should be created as [Single File Vue components](https://vuejs.org/v2/guide/single-file-components.html) and names using `CamelCase.vue`.
+All the component properties should be named using `camelCase` as well.
+All the component data fields should be `camelCased`.
+
+Theme catalog structure is well described in here: [Working with themes](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/themes/Working%20with%20themes.md)
+
+## Extensions
+For the reference of how extension can be structured please take a look at [template](https://github.com/DivanteLtd/vue-storefront/tree/develop/src/extensions/template).
+Check the [Working with extensions](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/extensions/Working%20with%20extensions.md) for more info.
+
+It's something like this:
+
+```
+|____store.js
+|____index.js
+|____components
+| |____ComponentExample.vue
+|____router.js
+|____package.json
+|____pages
+| |____PageExample.vue
+```
+
+## Events
+We're using [Vue events](https://vuejs.org/v2/guide/events.html) for async. operations handling and exposing some extension points in the code.
+
+The naming convention for the new events to be introduced is:
+`module-[before|after]-operation` - for example the event fired after order has been placed is named: `order-after-placed` and the event fired after product is loaded: `product-after-load`. 
+
+## Vuex
+We're using Vuex for all kind of state/date related operations. Vuex stores are modularized and Your extensions can have additional Vuex modules as well by just extending it's own [`store.js`](https://github.com/DivanteLtd/vue-storefront/tree/master/src/extensions/custom_extension).
+If You're adding some data operations to Core - check if t shouldn't be another module for that, then create the store separating the `actions`, `mutations` and `getters` to separate files:
+
+```
+.
+|____order
+| |____mutations.js
+| |____actions.js
+| |____index.js
+|____attribute
+| |____mutations.js
+| |____actions.js
+| |____getters.js
+| |____index.js
+| |____helpers.js
+|____homepage
+| |____index.js
+...
+```
+Check the [Working with Vuex](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/data/Working%20with%20Vuex.md) for more info.
+
+Please use Vuex for all state/data management and avoid loading local properties/local data inside Your components. This will casue problems with reactivity and SSR.
+
+## Data fetching
+All remote operations should've been placed inside the Vuex actions, using `fetch()` or exisiting [`data/sync`](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/data/Sync%20Store.md) module.
```

Comment:
I think we should recommend only one way dealing with external data, with all logic handled by VS core.

Reply:
Probably we need to take a look at the naming of this part, b/c for me `sync` = `synchronously` and I believe that most ppl will expect something like `fetch` or `get`, b/c we are not synchronizing the whole store with some external resource, but just asking about some particular set of data.

Reply:
Good point; I think we should change sync to fetch - I can do this

Reply:
sync = synchronization :D